### PR TITLE
Execute builds in a single BuildKit session (most pieces anyway), get rid of fakedep

### PR DESCRIPTION
**Pull request by vladaionescu**
Depends on https://github.com/earthly/buildkit/pull/11

* The net benefit of this PR is performance: for a one line go change in our codebase, the build used to take ~68s. Now it takes ~15s (~4.5X improvement).
* Merging all solves in a single session also fixes a number of obscure edge case bugs.

Changes:
* Perform main build, image outputs and artifact outputs within a single BuildKit session
* Eliminate fakedep and instead add all deps as refs to the response of the build function. Fakedep was extremely slow (assuming due to the need to rehash the FS over and over again)
* Order of printout of the outputs is now consistent between runs

Fixes https://github.com/earthly/earthly/issues/99
Fixes https://github.com/earthly/earthly/issues/85
Fixes https://github.com/earthly/earthly/issues/35
Fixes https://github.com/earthly/earthly/issues/18

TODO in future PRs:
* Some git operations are still using the old solves.
* Building a generated Dockerfile via `FROM DOCKERFILE` also still uses the old solves.
* WITH DOCKER --load and --pull also use the old solves.
* Execution of RUN --push also uses the old solves.

### CODE REVIEW

File: Earthfile [hunk 1]
```diff
@@ -24,6 +24,8 @@ deps:
     RUN go get golang.org/x/lint/golint
     RUN go get github.com/gordonklaus/ineffassign
     COPY go.mod go.sum ./
+    # @#
```
Comment (alexcb) [inline Earthfile:27]: is this comment intentional?

Comment (vladaionescu) [reply Earthfile]: Fixed :-)

File: buildkitd/Earthfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 
 buildkitd:
-    ARG BUILDKIT_BASE_IMAGE=github.com/earthly/buildkit:earthly-master+build
+    ARG BUILDKIT_BASE_IMAGE=github.com/earthly/buildkit:vlad/exporters+build
```
Comment (alexcb) [inline buildkitd/Earthfile:3]: can this be switched back now that the buildkit PR was merged?

Comment (vladaionescu) [reply buildkitd/Earthfile]: Yes done

**Review by alexcb — APPROVED** [review]
very nice!

**Comment by vladaionescu** [comment]
One of the tests flakes with `./e/tests+build-earth-test *failed* | ERROR: () failed to send last stat: EOF`. I encountered this issue locally too - I'll attempt to fix.

**Comment by vladaionescu** [comment]
> One of the tests flakes with ./e/tests+build-earth-test *failed* | ERROR: () failed to send last stat: EOF. I encountered this issue locally too - I'll attempt to fix.

I believe this has been resolved now.

**Pull request merged**